tests/phpunit/Authentication/OAuth20Test.php

```diff
@@ -4,55 +4,101 @@
+        $request1 = $history->pop()->getRequest();
+        self::assertSame('Bearer testToken', $request1->getHeaderLine('Authorization'));
+        self::assertSame('test', $request1->getHeaderLine('X-Test'));
+        // 2. request
+        self::assertEquals((object) ['foo' => 'bar2'], $restClient->download(new RestRequest(['endpoint' => 'ep'])));
+        $request2 = $history->pop()->getRequest();
+        self::assertSame('Bearer testToken', $request2->getHeaderLine('Authorization'));
+        self::assertSame('test', $request2->getHeaderLine('X-Test'));
+
+        // No more history items
+        self::assertTrue($history->isEmpty());
     }
 
     public function testMACAuth(): void
     {
+        // Create OAuth20 auth
         $config = json_decode((string) file_get_contents(__DIR__ . '/../data/oauth20mac/config.json'), true);
-        $restClient = new RestClient(new NullLogger(), ['base_url' => 'http://example.com'], [], []);
+        $authData = json_decode($config['authorization']['oauth_api']['credentials']['#data']);
         $auth = new OAuth20(
-            [],
             $config['authorization'],
             $config['parameters']['api']['authentication']
         );
-        $auth->attachToClient($restClient);
 
-        $request = $restClient->getClient()->createRequest('GET', '/resource?k=v');
-        self::sendRequest($restClient->getClient(), $request);
+        // Create RestClient
+        $history = new HistoryContainer();
+        $restClient = RestClientMockBuilder::create()
+            ->addResponse200('{"foo": "bar1"}')
+            ->addResponse200('{"foo": "bar2"}')
+            ->setGuzzleConfig(['headers' => ['X-Test' => 'test']])
+            ->setHistoryContainer($history)
+            ->setInitCallback(function (RestClient $restClient) use ($auth): void {
+                $auth->attachToClient($restClient);
+            })
+            ->getRestClient();
 
-        $authData = json_decode($config['authorization']['oauth_api']['credentials']['#data']);
+        // Each request contains Authorization header
+        // 1. request
+        self::assertEquals(
+            (object) ['foo' => 'bar1'],
+            $restClient->download(new RestRequest(['endpoint' => '/resource', 'params' => ['k1' => 'v1']]))
+        );
+        $this->assertMacRequest($history->pop()->getRequest(), $authData);
+        // 2. request
+        self::assertEquals(
+            (object) ['foo' => 'bar2'],
+            $restClient->download(new RestRequest(['endpoint' => '/resource', 'params' => ['k2' => 'v2']]))
+        );
+        $this->assertMacRequest($history->pop()->getRequest(), $authData);
+
+        // No more history items
+        self::assertTrue($history->isEmpty());
+    }
 
-        $authHeader = $request->getHeader('Authorization');
+    private function assertMacRequest(RequestInterface $request, \stdClass $authData): void
```

Comment:
OAuth20 "MAC" test, to iste, len je hlavcika a query zlozitejsia, je to opat cele zadane cez JSON:
`/../data/oauth20mac/config.json`